sphinxcontrib/jupyter/writers/translate_all.py

```diff
@@ -78,6 +83,13 @@ def visit_document(self, node):
         """
         JupyterCodeTranslator.visit_document(self, node)
 
+        ## if the source file parsed is index_toc and target is pdf
+        if "index_toc" in self.source_file_name and self.jupyter_target_pdf:
```

Comment:
we should replace `index_toc` with a variable from `conf.py` indicating which document contains the `toc` for book style PDF